Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -194,14 +199,43 @@ extension Configuration {
                 }
 
             case let .default(disabled: disabledRules, optIn: optInRules):
-                if rule is OptInRule.Type, Set(optInRules).isDisjoint(with: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is not enabled on " +
-                        "'\(Key.optInRules.rawValue)'.")
-                } else if Set(disabledRules).isSuperset(of: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is disabled on " +
-                        "'\(Key.disabledRules.rawValue)'.")
+                validateConfiguredRulesAreEnabled(
+                    message: message,
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    rule: rule
```

Comment:
Better to name `ruleType`:
```suggestion
                    ruleType: ruleType
```